students/eowyn/session04/revised_mailroom.py

```diff
@@ -0,0 +1,123 @@
+#!/usr/bin/env python
+
+import sys
+
+''' global variables'''
+donor_names = ["Margaret Atwood", "Fred Armisen",
+               "Heinz the Baron Krauss von Espy"]
+donations = [[300, 555], [240, 422, 1000], [1500, 2300]]
+DONORS = dict(zip(donor_names, donations))
+
+
+def mainloop():
+    ''' main interactive loop 
+    "send a thank you" "create a report" or "quit"
+    '''
+    while True:
+        answer = input("Select from one of these options: \n"
+                       "(1) Send a Thank You\n"
+                       "(2) Create a Report\n"
+                       "(3) Send letters to everyone\n"
+                       "(4) Quit\n>")
+        arg_dict = {"1": thank_you,
+                    "2": print_report,
+                    "3": send_letters,
+                    "4": quit_code}
+        try:
+            arg_dict.get(answer)()
+        except TypeError:
+            continue
+
+
+def quit_code():
+    sys.exit()
+
+
+def thank_you():
+    ''' Update donor records and print thank you notes.'''
+    while True:
+        fullname = input("To send a thank you, "
+                         "Enter a donor name (new or existing)\n"
+                         "Enter all to list existing DONORS \n"
+                         "Enter menu to return to main menu \n >")
+        fullname = remove_inputquotes(fullname)
+        if fullname == 'menu':
+            break
+        if fullname.lower() == 'all':
```

Comment:
This isn't a LOT of logic, but still probably better to put it in individual functions.

why will become clear next week when we write unit tests for the whole thing.